### PR TITLE
CORE-17310 - Show same member in any status from member lookup API

### PR DESCRIPTION
**Pull request by dimosr**
## Changes
Adjusting the member lookup API to return the member on behalf of which the lookup is performed regardless of their status. This will allow members to always be able to check their status. The old version (v1) of the API retains the existing behaviour to ensure this change is not breaking.

Existing unit tests for v1 were copied for v5_1 and a new test case was added for the case where the member performing the lookup is suspended.

## Testing
Apart from the additional unit tests, I did some manual e2e testing using the combined worker. Running the `SingleClusterDynamicNetworkTest` plus an additional suspension for Alice, I get the following results from member lookup on behalf of Alice:
* v1 API (Alice not returned):
```
{
  "members": [
    {
      "memberContext": {
        "corda.cpi.name": "mgm.cpi",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.ecdh.key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE9b2AIXKJ+338cmNvUvlF83RlEnXv\nAilnMyJB3WthSAr3Zn49InGBXTWPEPwUx8UwWWdsPewOQ9j/Y8CuyGbZJA==\n-----END PUBLIC KEY-----\n",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.name": "OU=c4382bb3-edc6-4af9-8b6c-b7015e38e87a, O=Mgm, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.session.keys.0.hash": "SHA-256:98CD2A66D870CE9F7B888944B1FAD420A707BD3FCDE282648938F63416653ABA",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEndaFJdke3PrWAHodkbKa+0QXX/m2\nVfmadua+Uwg7IDZo+UwdH/IbuWex6ec9utBCUzwbIhYpLYQAk960fqKStQ==\n-----END PUBLIC KEY-----\n",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:48:53.452313Z",
        "corda.mgm": "true",
        "corda.modifiedTime": "2023-10-04T08:48:53.452313Z",
        "corda.serial": "1",
        "corda.status": "ACTIVE"
      }
    },
    {
      "memberContext": {
        "corda.cpi.name": "c4382bb3-edc6-4af9-8b6c-b7015e38e87a",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.ledger.keys.0.hash": "SHA-256:CA152055E8305FC8E362FB9444D5D23285BB21821C39F80042C93C50A41FD9AF",
        "corda.ledger.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEX2odA/+LGI8CoSHPLmezkkUE0l5i\nzWUYWSBDU9qMt6GxjeK8JeyDZgzIIuglGOneMEOxHhAgovzGt0XTVOw07w==\n-----END PUBLIC KEY-----\n",
        "corda.ledger.keys.0.signature.spec": "SHA256withECDSA",
        "corda.name": "CN=Bob-c4382bb3-edc6-4af9-8b6c-b7015e38e87a, OU=Application, O=R3, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.registration.id": "4094dbf0-c5b4-41ef-a76d-c78d61102122",
        "corda.session.keys.0.hash": "SHA-256:F98A0A1B83B8015FC72C29F6E087153572F5223E07586BC975030548BFE6D3AB",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWrfU84W6riI41V50lsxSrWHqe8K1\nLtKtjveRQAanoO+fGjdYVTAmDdlnme+9+DLsoYHNDXjAPE9zIIKHxt5uNQ==\n-----END PUBLIC KEY-----\n",
        "corda.session.keys.0.signature.spec": "SHA256withECDSA",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:50:17.070274Z",
        "corda.modifiedTime": "2023-10-04T08:50:17.070274Z",
        "corda.serial": "1",
        "corda.status": "ACTIVE"
      }
    },
    {
      "memberContext": {
        "corda.cpi.name": "c4382bb3-edc6-4af9-8b6c-b7015e38e87a",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.ledger.keys.0.hash": "SHA-256:F8E32077EFCF046BAEC6378319269B4C2762A4B08408EFACC76018906B3B3B8F",
        "corda.ledger.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEDcn7yjwSsiLyXblllLy2qP20oQN0\nY2UdZsQy79CAKVdVDVrxqIvdbDQEzJtFn0DxL7+pUn4mtUUtsSv9nTz3XA==\n-----END PUBLIC KEY-----\n",
        "corda.ledger.keys.0.signature.spec": "SHA256withECDSA",
        "corda.name": "CN=Notary-c4382bb3-edc6-4af9-8b6c-b7015e38e87a, OU=Application, O=R3, L=London, C=GB",
        "corda.notary.keys.0.hash": "SHA-256:81A9BD14AAAC4E2F03D2E6BA23E2A741A44212882EF430E4CE4E3FC46C525DF1",
        "corda.notary.keys.0.id": "81A9BD14AAAC",
        "corda.notary.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEX4tXml46X5nQYDChjTUqXE1anuKX\n1iiO3ODXuBgzCAcrNzzKP5/2tjgaI76bZrWAxBgrsqoXf9YVUHXusxCKDw==\n-----END PUBLIC KEY-----\n",
        "corda.notary.keys.0.signature.spec": "SHA256withECDSA",
        "corda.notary.service.flow.protocol.name": "com.r3.corda.notary.plugin.nonvalidating",
        "corda.notary.service.flow.protocol.version.0": "1",
        "corda.notary.service.name": "O=NotaryService, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.registration.id": "c8142f69-5874-4c07-a963-7e759af64b0e",
        "corda.roles.0": "notary",
        "corda.session.keys.0.hash": "SHA-256:8FBFD2A671F91008893E39F457BBD0F9FF0EB1579F901012AE50772C22EFC763",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAExpVjqZiDG/onlwgylB335xuVEL8t\nc51W5R6cWvP2xDsxUnZEYZ51H1DJjEGfJRDkj7v3CVlxOtX95Ial4hSPNQ==\n-----END PUBLIC KEY-----\n",
        "corda.session.keys.0.signature.spec": "SHA256withECDSA",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:51:15.124195Z",
        "corda.modifiedTime": "2023-10-04T08:51:15.124195Z",
        "corda.serial": "1",
        "corda.status": "ACTIVE"
      }
    }
  ]
}
```
* v5_1 API (Alice returned):
```
{
  "members": [
    {
      "memberContext": {
        "corda.cpi.name": "mgm.cpi",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.ecdh.key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE9b2AIXKJ+338cmNvUvlF83RlEnXv\nAilnMyJB3WthSAr3Zn49InGBXTWPEPwUx8UwWWdsPewOQ9j/Y8CuyGbZJA==\n-----END PUBLIC KEY-----\n",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.name": "OU=c4382bb3-edc6-4af9-8b6c-b7015e38e87a, O=Mgm, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.session.keys.0.hash": "SHA-256:98CD2A66D870CE9F7B888944B1FAD420A707BD3FCDE282648938F63416653ABA",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEndaFJdke3PrWAHodkbKa+0QXX/m2\nVfmadua+Uwg7IDZo+UwdH/IbuWex6ec9utBCUzwbIhYpLYQAk960fqKStQ==\n-----END PUBLIC KEY-----\n",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:48:53.452313Z",
        "corda.mgm": "true",
        "corda.modifiedTime": "2023-10-04T08:48:53.452313Z",
        "corda.serial": "1",
        "corda.status": "ACTIVE"
      }
    },
    {
      "memberContext": {
        "corda.cpi.name": "c4382bb3-edc6-4af9-8b6c-b7015e38e87a",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.ledger.keys.0.hash": "SHA-256:CA152055E8305FC8E362FB9444D5D23285BB21821C39F80042C93C50A41FD9AF",
        "corda.ledger.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEX2odA/+LGI8CoSHPLmezkkUE0l5i\nzWUYWSBDU9qMt6GxjeK8JeyDZgzIIuglGOneMEOxHhAgovzGt0XTVOw07w==\n-----END PUBLIC KEY-----\n",
        "corda.ledger.keys.0.signature.spec": "SHA256withECDSA",
        "corda.name": "CN=Bob-c4382bb3-edc6-4af9-8b6c-b7015e38e87a, OU=Application, O=R3, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.registration.id": "4094dbf0-c5b4-41ef-a76d-c78d61102122",
        "corda.session.keys.0.hash": "SHA-256:F98A0A1B83B8015FC72C29F6E087153572F5223E07586BC975030548BFE6D3AB",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWrfU84W6riI41V50lsxSrWHqe8K1\nLtKtjveRQAanoO+fGjdYVTAmDdlnme+9+DLsoYHNDXjAPE9zIIKHxt5uNQ==\n-----END PUBLIC KEY-----\n",
        "corda.session.keys.0.signature.spec": "SHA256withECDSA",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:50:17.070274Z",
        "corda.modifiedTime": "2023-10-04T08:50:17.070274Z",
        "corda.serial": "1",
        "corda.status": "ACTIVE"
      }
    },
    {
      "memberContext": {
        "corda.cpi.name": "c4382bb3-edc6-4af9-8b6c-b7015e38e87a",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.ledger.keys.0.hash": "SHA-256:F8E32077EFCF046BAEC6378319269B4C2762A4B08408EFACC76018906B3B3B8F",
        "corda.ledger.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEDcn7yjwSsiLyXblllLy2qP20oQN0\nY2UdZsQy79CAKVdVDVrxqIvdbDQEzJtFn0DxL7+pUn4mtUUtsSv9nTz3XA==\n-----END PUBLIC KEY-----\n",
        "corda.ledger.keys.0.signature.spec": "SHA256withECDSA",
        "corda.name": "CN=Notary-c4382bb3-edc6-4af9-8b6c-b7015e38e87a, OU=Application, O=R3, L=London, C=GB",
        "corda.notary.keys.0.hash": "SHA-256:81A9BD14AAAC4E2F03D2E6BA23E2A741A44212882EF430E4CE4E3FC46C525DF1",
        "corda.notary.keys.0.id": "81A9BD14AAAC",
        "corda.notary.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEX4tXml46X5nQYDChjTUqXE1anuKX\n1iiO3ODXuBgzCAcrNzzKP5/2tjgaI76bZrWAxBgrsqoXf9YVUHXusxCKDw==\n-----END PUBLIC KEY-----\n",
        "corda.notary.keys.0.signature.spec": "SHA256withECDSA",
        "corda.notary.service.flow.protocol.name": "com.r3.corda.notary.plugin.nonvalidating",
        "corda.notary.service.flow.protocol.version.0": "1",
        "corda.notary.service.name": "O=NotaryService, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.registration.id": "c8142f69-5874-4c07-a963-7e759af64b0e",
        "corda.roles.0": "notary",
        "corda.session.keys.0.hash": "SHA-256:8FBFD2A671F91008893E39F457BBD0F9FF0EB1579F901012AE50772C22EFC763",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAExpVjqZiDG/onlwgylB335xuVEL8t\nc51W5R6cWvP2xDsxUnZEYZ51H1DJjEGfJRDkj7v3CVlxOtX95Ial4hSPNQ==\n-----END PUBLIC KEY-----\n",
        "corda.session.keys.0.signature.spec": "SHA256withECDSA",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:51:15.124195Z",
        "corda.modifiedTime": "2023-10-04T08:51:15.124195Z",
        "corda.serial": "1",
        "corda.status": "ACTIVE"
      }
    },
    {
      "memberContext": {
        "corda.cpi.name": "c4382bb3-edc6-4af9-8b6c-b7015e38e87a",
        "corda.cpi.signer.summary.hash": "SHA-256:E53DA5C67A637D42335808FA1534005281BAE7E49CCE8833213E58E0FDCA8B35",
        "corda.cpi.version": "1.0.0.0-SNAPSHOT",
        "corda.endpoints.0.connectionURL": "https://localhost:8080",
        "corda.endpoints.0.protocolVersion": "1",
        "corda.groupId": "d0427782-b364-4e6b-814d-e76d8fa4fed2",
        "corda.ledger.keys.0.hash": "SHA-256:A2AA5DB8AF073EF1D09EA85D2AD191E19A93AE0EEFA4DC7FFF4959156430E9B2",
        "corda.ledger.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhLk8PjWbrJ3kn+EeGGhFhOHrI1kZ\nYrdS31V392ELcfssgnehPvXqD3xwDCQQyt6miqUbKFlRKgH76sR9YRmbCg==\n-----END PUBLIC KEY-----\n",
        "corda.ledger.keys.0.signature.spec": "SHA256withECDSA",
        "corda.name": "CN=Alice-c4382bb3-edc6-4af9-8b6c-b7015e38e87a, OU=Application, O=R3, L=London, C=GB",
        "corda.platformVersion": "50100",
        "corda.registration.id": "ef6f65bf-8c14-46c4-8959-495ae1ce6842",
        "corda.session.keys.0.hash": "SHA-256:15B2893796E9D4EC3633FBC629BED2D64E5692220A8C10A622179B989E3E6187",
        "corda.session.keys.0.pem": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEmf8TbXMTsnMjnAq0ClJj/vR+i+sK\nLX4on+SsAaNA1sgyqRnBhbeUUgGrFmZUhJZMqO4U2rQrR4LJ2Nwgfxbsag==\n-----END PUBLIC KEY-----\n",
        "corda.session.keys.0.signature.spec": "SHA256withECDSA",
        "corda.softwareVersion": "5.1.0.0-SNAPSHOT"
      },
      "mgmContext": {
        "corda.creationTime": "2023-10-04T08:49:27.118695Z",
        "corda.modifiedTime": "2023-10-04T08:51:44.856564Z",
        "corda.serial": "2",
        "corda.status": "SUSPENDED"
      }
    }
  ]
}
```

### CODE REVIEW

**Review by vkolomeyko — DISMISSED** [review]
LGTM

File: components/membership/membership-rest/src/main/kotlin/net/corda/membership/rest/v1/MemberLookupRestResource.kt [hunk 1]
```diff
@@ -42,12 +43,14 @@ interface MemberLookupRestResource : RestResource {
...
      * @return List of active and pending members matching the criteria as [RestMemberInfoList].
      */
     @HttpGET(
         path = "{holdingIdentityShortHash}",
-        description = "This method retrieves a list of all active and pending members in the membership group."
+        description = "This method retrieves a list of all active and pending members in the membership group.",
+        maxVersion = RestApiVersion.C5_0
     )
     @Suppress("LongParameterList")
     fun lookup(
```
Comment (williamvigorr3) [inline components/membership/membership-rest/src/main/kotlin/net/corda/membership/rest/v1/MemberLookupRestResource.kt:57]: I think this should be deprecated. https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4472635529/Versioning+of+REST+API#Terminating-the-old-method-at-a-particular-version

**Review by williamvigorr3 — COMMENTED** [review]
I left a comment. Do we need to change an E2E tests to test the old the endpoint?

**Comment by dimosr** [comment]
> Do we need to change an E2E tests to test the old the endpoint?

The unit tests I've added provide quite a good coverage on the old API in my opinion, so I don't think it would be very risky not adding e2e test about that (and it would save some test time/resources). But, I don't have strong opinions so if you believe we should e2e test it too, happy to do that.

**Comment by williamvigorr3** [comment]
> > Do we need to change an E2E tests to test the old the endpoint?
> 
> The unit tests I've added provide quite a good coverage on the old API in my opinion, so I don't think it would be very risky not adding e2e test about that (and it would save some test time/resources). But, I don't have strong opinions so if you believe we should e2e test it too, happy to do that.

I also think it's fine (judging by how this all works). However https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4472635529/Versioning+of+REST+API#Changing-E2E%2C-Smoke-and-performance-tests says to test it.

**Review by williamvigorr3 — DISMISSED** [review]
LGTM

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Comment by dimosr** [comment]
Thanks for the reviews.

FYI, I am working on a PR with the necessary adjustments to the e2e tests here: https://github.com/corda/corda-e2e-tests/pull/264

I'll try and get that approved first, before merging this one. So, that they can be merged in quick succession and avoid long-running breakages in the e2e tests.

**Review by vkolomeyko — APPROVED** [review]

**Pull request merged**